pkg/datastore/datastore.go

```diff
@@ -751,21 +752,32 @@ type Stats struct {
 	ObjectTypeStatistics []ObjectTypeStat
 }
 
-// RelationshipIterator is an iterator over matched tuples.
-type RelationshipIterator interface {
-	// Next returns the next tuple in the result set.
-	Next() *core.RelationTuple
+// RelationshipIterator is an iterator over matched tuples. It is a single use
+// iterator.
+type RelationshipIterator iter.Seq2[tuple.Relationship, error]
 
-	// Cursor returns a cursor that can be used to resume reading of relationships
-	// from the last relationship returned. Only applies if a sort ordering was
-	// requested.
-	Cursor() (options.Cursor, error)
+func IteratorToSlice(iter RelationshipIterator) ([]tuple.Relationship, error) {
```

Comment:
I think these are only used in tests, but they could use some basic unit tests

Reply:
Yeah, they are only for tests. They are so simple I'm not sure we need a unit test for them; maybe we move them into util as generic and then test there?

Reply:
Let's do that as a follow up